node/src/components/linear_chain_sync/operations.rs

```diff
@@ -312,6 +276,73 @@ where
     Ok(None)
 }
 
+pub(super) struct SwitchBlockInfo {
```

Comment:
I'd call this `KeyBlockInfo`, so it's clearer that the fields refer to the _next_ era.
(Alternatively, rename the fields to `next_era_validator_weights` and `timestamp`.)

Reply:
Fixed in a7a3a3118